PowerValidatedSolutions.psm1

```diff
@@ -50441,6 +50441,469 @@ Function Test-WMSubnetInput {
+    }
+    if ($expireDays -lt 2) {
+        Write-Error "Certificate expiration day ($expireDays) is less than 1 day.  Please provide a value greater than 1 day."
+    }
+
+    # Create temporary cfg file
+    $cfgString = '[ req ]
+    days = ' +$expireDays+ '
+    default_md = sha512
+    default_bits = ' +$keySize+ '
+    distinguished_name = req_distinguished_name
+    prompt = no
+    req_extensions = v3_req
+    
+    [ v3_req ]
+    basicConstraints = CA:FALSE
+    keyUsage = digitalSignature, keyEncipherment, dataEncipherment, nonRepudiation
+    extendedKeyUsage = serverAuth, clientAuth
+    subjectAltName = @alternate_names
+
+    [ req_distinguished_name ]
+    commonName = ' +$commonName+ '
+    countryName = ' +$country+ '
+    stateOrProvinceName = ' +$state+ '
+    localityName = ' +$locality+ '
+    0.organizationName = ' +$organization+ '
+    organizationalUnitName = ' +$organizationUnit+ '
+    
+    [ alternate_names ]
+    '
+    # Parsing SAN string
+    $sanArrary = $subjectAlternativeNames -split "\s*,\s*"
+    $sanCount = 1
+
+    foreach($san in $sanArrary) {
+        $cfgString = $cfgString + "DNS." + $sanCount + " = " + $san + "`n"
+        $sanCount++
+    }
+
+    # Generate RSA private key and CSR file
+    $cfgString | Out-File $cfgFilePath
+    openssl req -new -sha256 -nodes -out $csrFilePath -keyout $orginalKeyFilePath -config $cfgFilePath >> $null 2>$null
+    openssl rsa -in $orginalKeyFilePath -out $keyFilePath -traditional >> $null 2>$null
+
+    # Clean up
+    Remove-Item $cfgFilePath
+    Remove-Item $orginalKeyFilePath
+    
+}
+Export-ModuleMember -Function Invoke-GeneratePrivateKeyAndCsr
+
+Function Invoke-RequestSignedCertificate {
+    <#
+        .SYNOPSIS
+        Request Certificate Signing Request (CSR) to be signed by a Microsoft Certificate Authority.
+
+        .DESCRIPTION
+        The Invoke-RequestSignedCertificate cmdlet connects to Microsoft Certificate Authority web enrollment service to sign a certificate signing request (CSR).
+        
+        .EXAMPLE
+        Invoke-RequestSignedCertificate -caFqdn "rpl-ad01.rainpole.io" -csrFilePath "C:\certificates\sfo-vrli01.rainpole.io.csr" -certDirPath "C:\certificates" -certificateAuthority "msca" -username "Administrator" -password "VMw@re1!" -certificateTemplate "VMware"
+        This example will request the sfo-vrli01 CSR file to be signed by the Certificate Authority rpl-ad01.rainpole.io.
+
+        .EXAMPLE
+        Invoke-RequestSignedCertificate -caFqdn "rpl-ad01.rainpole.io" -csrFilePath "C:\certificates\sfo-vrli01.rainpole.io.csr" -certDirPath "C:\certificates" -certificateAuthority "openssl" -caKeyPath "C:\certificates\CAroot.key" -caCertPath "C:\certificates\CAroot.pem" -expireDays 365
+        This example will request the sfo-vrli01 CSR file to be signed using openSSL and provided the Certificate Authority key and certificate.
```

Comment:
```suggestion
        This example will request the sfo-vrli01 CSR file to be signed using OpenSSL and provided the Certificate Authority key and certificate.
```

Check to use of "OpenSSL", the componnet's correct name / casing.

Reply:
addressed in commit cfcd77a79eb20278e44b123b0092c173dd7b5014